src/engine/bpmcontrol.cpp

```diff
@@ -850,7 +850,7 @@ void BpmControl::collectFeatures(GroupFeatureState* pGroupFeatures) const {
                        &dThisBeatLength, &dThisBeatFraction)) {
         pGroupFeatures->has_beat_length_sec = true;
         // Note: dThisBeatLength is fractional frames count * 2 (stereo samples)  
-        pGroupFeatures->beat_length_sec = dThisBeatLength / m_pTrack->getSampleRate() / 2 * calcRateRatio();
+        pGroupFeatures->beat_length_sec = dThisBeatLength / m_pTrack->getSampleRate() / 2 / calcRateRatio();
```

Comment:
The many division operators are difficult to read.

Enclosing in parentheses would at least isolate the integer division:
```
(m_pTrack->getSampleRate() / 2)
```

kNumberOfChannels = 2 should be defined as a constexpr + comment instead of hiding it somewhere in the code.